src/redux/reducers/greetingSlice.js

```diff
@@ -0,0 +1,32 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const initialState = { message: '' };
+
+export const fetchRandomGreeting = createAsyncThunk(
+  'greeting/fetchRandomGreeting',
+  async () => {
+    const response = await fetch('http://localhost:3001/random_greeting');
+    const data = await response.json();
+    return data.message;
+  },
+);
+
+const greetingSlice = createSlice({
+  name: 'greeting',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchRandomGreeting.pending, (state) => {
+        state.message = 'Loading...';
+      })
+      .addCase(fetchRandomGreeting.fulfilled, (state, action) => {
+        state.message = action.payload;
+      })
+      .addCase(fetchRandomGreeting.rejected, (state) => {
+        state.message = 'Error fetching greeting';
+      });
+  },
+});
+
+export default greetingSlice.reducer;
```

Comment:
- [ ] [REQUIRED] The frontend is rendering this instead of rending the greetings coming from the backend. Let's fix this error, please.
<img width="408" alt="Screenshot 2024-02-17 at 21 15 40" src="https://github.com/MNisarAli/hello-react-front-end/assets/58692590/bdd79e68-111e-4e1e-b36b-e77597c0ddb6">

Reply:
Hi! :wave:  @Whoistolu, thank you for your valuable review. I appreciate your feedback and have made corrections to fix the error.

Reply:
@MNisarAli , Have you pushed your latest changes? I am your newest code reviewer.

Reply:
@topeogunleye Yes, I have pushed the changes. The problem was in the backend. Now, I have fixed that by setting up rack-cors configuration.

Reply:
Error still persists:

<img width="273" alt="image" src="https://github.com/MNisarAli/hello-react-front-end/assets/52934128/d9bd8040-d9eb-44f3-b76d-58ea86e08cea">

Reply:
Bro the error is fixed if you please run the app on localhost:3000 port